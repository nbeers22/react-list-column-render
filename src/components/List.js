import React from 'react';
import { STORE } from '../store.js';
import Card from './Card.js'

const List = props => {

  const listCards = props.cards.map( (cardID,index) => {
    if(STORE.allCards.hasOwnProperty(cardID)){
      return <Card key={index} title={STORE.allCards[cardID].title} content={STORE.allCards[cardID].content} />
    }
  });

  return (
    <section className="List">
      <header className="List-header">
        <h2>{props.header}</h2>
      </header>
      <div className="List-cards">
        {listCards}
      </div>
    </section>
  )
}

export default List;