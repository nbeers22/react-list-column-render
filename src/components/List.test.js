import React from 'react'
import ReactDOM from 'react-dom'
import List from './List.js'
import renderer from 'react-test-renderer'
import { exportAllDeclaration } from '@babel/types';

describe('List Component', () => {

  // Smoke Test
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<List cards={['a','b']} header="List 1" />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  // Snapshot Test
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<List cards={['a','b','c','d']} header="List 1" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the UI as expected with no header prop passed', () => {
    const tree = renderer
      .create(< List cards={['a','b']} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });


});