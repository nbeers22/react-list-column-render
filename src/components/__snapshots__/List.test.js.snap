// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      List 1
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Third card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Fourth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
</section>
`;

exports[`List Component renders the UI as expected with no header prop passed 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2 />
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
</section>
`;
